sdk/storage/faq.md

```diff
@@ -0,0 +1,374 @@
+
+[How to troubleshoot 403 errors?](#how-to-troubleshoot-403-errors)
+
+## How to use list-operations? Why doesn't list-operation return all blobs in a container?
+
+C++ SDK applies a different design for pageable operations from other languages.
+It involves a nested-loop to iterate over all items in a container.
+The outer loop does network I/O operations.
+The inner loop gets called for every paged result and doesn't do I/O.
+
+Below is an example of listing all blobs in a blob container.
+
+```C++
+for (auto page = blobContainerClient.ListBlobs(); page.HasPage(); page.MoveToNextPage()) {
+  for (auto& blob : page.Blobs) {
+    std::cout << blob.Name << std::endl;
+  }
+}
+```
+
+Sometimes a paged result may contain multiple collections and you may want to iterate over all of them.
+
+```C++
+for (auto page = directoryClient.ListFilesAndDirectories(); page.HasPage(); page.MoveToNextPage())
+{
+  for (const auto& d : page.Directories)
+  {
+    std::cout << "Directory: " << d.Name << std::endl;
+  }
+  for (const auto& f : page.Files)
+  {
+    std::cout << "File: " << f.Name << std::endl;
+  }
+}
+```
+
+## Can I specify an API-version other than the default one?
+
+Yes, each client options takes an `ApiVersion` as optional parameter, with which you can specify an API-version used for all the HTTP requests from this client.
+Client spawned from another client instance will inherit the settings.
+
+```C++
+// serviceClient sends HTTP requests with default API-version, which will change as version evolves.
+auto serviceClient = BlobServiceClient::CreateFromConnectionString(GetConnectionString());
+
+// serviceClient sends HTTP requests with specified API-version.
+BlobClientOptions clientOptions;
+clientOptions.ApiVersion = "2019-12-12";
+auto serviceClient = BlobServiceClient::CreateFromConnectionString(GetConnectionString, clientOptions);
+// containerClient inherits this setting.
+auto containerClient = serviceClient.GetBlobContainerClient(containerName);
+```
+
+Please note that multiple-version is not supported yet by Storage C++ SDK (This may change in the future).
+Which means, the code above only changes the `x-ms-version` request header and doesn't change SDK behavior.
+SDK will still set headers and query parameters that are added between the customized API version (which is usually older) and latest API version,
+and still expect some headers or fields in HTTP response.
+Furthermore, this scenario is not covered by testing, although most of the APIs should be able to work. You should do thorough testing before your code goes to production.
+
+## How can I set a custom UserAgent string?
+
+We recommend you set an application ID with code below, so that it can be identified from which application, SDK, platform the request was sent. The information could be useful for troubleshooting and telemetry purposes.
```

Comment:
```suggestion
We recommend you set an application ID with the code below, so that it can be identified from which application, SDK, and platform the request was sent. The information could be useful for troubleshooting and telemetry purposes.
```